.editorconfig

```diff
@@ -0,0 +1,13 @@
+# Editor configuration, see https://editorconfig.org
+root = true
+
+[*]
+charset = utf-8
+indent_style = space
+indent_size = 2
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+[*.md]
+max_line_length = off
+trim_trailing_whitespace = false
```

Comment:
Why not trim trailing whitespace?

Reply:
Can you move this to `/linux/.editorconfig`?

This way, each platform can define their own (I've already got `/android/.editorconfig`)

Reply:
@mcdurdin I think there was a reason why I added this (in a different project), but unfortunately I don't remember the details. Or maybe it was just so that it keeps the trailing space at the end
of a line which makes it easier to join lines if you reformat a paragraph.

@darcywong00 each platform can still define their own. Editconfig supports having multiple `.editorconfig` files in different subdirectories. If there are conflicting values the one closer in the file/directory structure wins. 

IMO it makes sense to have a `.editorconfig` file at the top level that defines some settings so that not every platform has to redefine it. The different platforms can override the values if they prefer/need different settings.

As it is right now, `/android/.editorconfig` has a `root = true` line. This causes editorconfig to stop looking in the parent directories, so this change wouldn't change anything for Android. It might make sense to remove the `root = true` line and just override the values where necessary.

Reply:
> Or maybe it was just so that it keeps the trailing space at the end
> of a line which makes it easier to join lines if you reformat a paragraph.

I see. I would prefer to trim whitespace on .md files I think.

Reply:
Done in #2336